soccer/MainWindow.cpp

```diff
@@ -718,8 +703,54 @@ void MainWindow::updateStatus() {
     RJ::Time curTime = RJ::now();
 
     // Determine if we are receiving packets from an external referee
-    bool haveExternalReferee =
-        (curTime - ps.lastRefereeTime) < RJ::Seconds(0.5);
+    bool haveExternalReferee = (curTime - ps.lastRefereeTime) < RJ::Seconds(1);
+
+    std::vector<int> validIds = _processor->state()->ourValidIds();
+
+    for (int i = 1; i < 17; i++) {
+        // This will need to change if the max number of robots changes
+        if (std::find(validIds.begin(), validIds.end(), i - 1) !=
+            validIds.end()) {
+            // The list starts with None so i is 1 higher than the shell id
+            _ui.goalieID->setItemData(i, true, Qt::UserRole);
+        } else {
+            _ui.goalieID->setItemData(i, false, Qt::UserRole - 1);
+        }
+    }
+
+    if (haveExternalReferee && _autoExternalReferee) {
+        // External Ref is connected and should be used
+        _ui.fastHalt->setEnabled(false);
+        _ui.fastStop->setEnabled(false);
+        _ui.fastReady->setEnabled(false);
+        _ui.fastForceStart->setEnabled(false);
+        _ui.fastKickoffBlue->setEnabled(false);
+        _ui.fastKickoffYellow->setEnabled(false);
+    } else {
+        _ui.fastHalt->setEnabled(true);
+        _ui.fastStop->setEnabled(true);
+        _ui.fastReady->setEnabled(true);
+        _ui.fastForceStart->setEnabled(true);
+        _ui.fastKickoffBlue->setEnabled(true);
+        _ui.fastKickoffYellow->setEnabled(true);
+    }
+
+    if (std::find(validIds.begin(), validIds.end(),
+                  _processor->state()->gameState.getGoalieId()) !=
+            validIds.end() &&
+        haveExternalReferee) {
+        // The External Ref is connected and transmitting a valid goalie ID
+        _ui.goalieID->setEnabled(false);
+
+        // Changes the goalie INDEX which is 1 higher than the goalie ID
+        if (_ui.goalieID->currentIndex() !=
+            _processor->state()->gameState.getGoalieId() + 1) {
+            _ui.goalieID->setCurrentIndex(
+                _processor->state()->gameState.getGoalieId() + 1);
+        }
+    } else {
+        _ui.goalieID->setEnabled(true);
+    }
 
     /*if (_autoExternalReferee && haveExternalReferee &&
     !_ui.externalReferee->isChecked())
```

Comment:
:+1:, yup, I was gonna ask you that when I finished the reiew, that would be perfect! :smile: